### DISCUSSION TITLE
questions about makeEnvPublic and EnvScript

### DISCUSSION THREAD

**Discussion by aNyMoRe0505** [expatfile/next-runtime-env#105] [Q&A]
Hi, thank you for creating this library. I have a few small questions I'd like to clarify. 

makeEnvPublic seems to be used when you want to expose some server variables to the public.
So, you can use `makeEnvPublic('FOO')` and then in the client component use `env('NEXT_PUBLIC_FOO')`.

My question is, can't we achieve the same effect directly with EnvScript like this:

```js
<EnvScript
  env={{
    NEXT_PUBLIC_FOO: process.env.FOO,
  }}
/>
```

And then in the client component just `env('NEXT_PUBLIC_FOO')`

So, why do we still need makeEnvPublic ?

**Comment by HofmannZ** [comment]
It results in the same and both should work fine. We prefer `makeEnvPublic` because we can tuck it away in our `instrumentation.ts`. Additionally, `makeEnvPublic` is slightly fewer lines of code ;)
